File: frontend/src/types/report.ts
export type ReportReason =
  | 'WRONG_TRANSLATION'
  | 'WRONG_CONTEXT'
  | 'WRONG_SPELLING'
  | 'INAPPROPRIATE_CONTENT'
  | 'OTHER';

export type ReportStatus = 'PENDING' | 'REVIEWED' | 'RESOLVED' | 'DISMISSED';

export interface WordReport {
  id: number;
  word: string;
  movie_id?: number;
  movie_title?: string;
  reason: ReportReason;
  details?: string;
  status: ReportStatus;
  reporter_id: number;
  reporter_email?: string;
  reviewer_id?: number;
  reviewer_email?: string;
  review_notes?: string;
  created_at: string;
  updated_at: string;
}

export interface ReportStats {
  pending: number;
  reviewed: number;
  resolved: number;
  dismissed: number;
  total: number;
}

export const REPORT_REASON_LABELS: Record<ReportReason, string> = {
  WRONG_TRANSLATION: 'Wrong translation',
  WRONG_CONTEXT: "Doesn't match context",
  WRONG_SPELLING: 'Wrong spelling',
  INAPPROPRIATE_CONTENT: 'Inappropriate content',
  OTHER: 'Other issue',
};

export const REPORT_STATUS_LABELS: Record<ReportStatus, string> = {
  PENDING: 'Pending',
  REVIEWED: 'Reviewed',
  RESOLVED: 'Resolved',
  DISMISSED: 'Dismissed',
};

File: backend/src/schemas/report.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from prisma.enums import reportreason, reportstatus


class ReportCreate(BaseModel):
    word: str
    movie_id: Optional[int] = None
    movie_title: Optional[str] = None
    reason: reportreason
    details: Optional[str] = None


class ReportUpdate(BaseModel):
    status: reportstatus
    review_notes: Optional[str] = None


class ReportResponse(BaseModel):
    id: int
    word: str
    movie_id: Optional[int] = None
    movie_title: Optional[str] = None
    reason: reportreason
    details: Optional[str] = None
    status: reportstatus
    reporter_id: int
    reporter_email: Optional[str] = None
    reviewer_id: Optional[int] = None
    reviewer_email: Optional[str] = None
    review_notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

    @classmethod
    def from_prisma(cls, report, include_relations=True):
        """Convert Prisma WordReport to ReportResponse"""
        data = {
            'id': report.id,
            'word': report.word,
            'movie_id': report.movieId,
            'movie_title': report.movieTitle,
            'reason': report.reason,
            'details': report.details,
            'status': report.status,
            'reporter_id': report.reporterId,
            'reviewer_id': report.reviewerId,
            'review_notes': report.reviewNotes,
            'created_at': report.createdAt,
            'updated_at': report.updatedAt,
        }

        if include_relations:
            data['reporter_email'] = report.reporter.email if report.reporter else None
            data['reviewer_email'] = report.reviewer.email if report.reviewer else None

        return cls(**data)


class ReportStats(BaseModel):
    pending: int
    reviewed: int
    resolved: int
    dismissed: int
    total: int
